File: server/app/providers/adapters/openai.py
from __future__ import annotations

from langchain_openai import ChatOpenAI

from ..base import ProviderRuntimeConfig, ProviderType


class OpenAIAdapter:
    provider_type: ProviderType = "openai"

    def validate(self, config: ProviderRuntimeConfig) -> None:
        if config.model.strip() == "":
            raise ValueError("openai provider requires model")
        if config.api_key is None or config.api_key.strip() == "":
            raise ValueError("openai provider requires api_key")

    def create_model(self, config: ProviderRuntimeConfig) -> ChatOpenAI:
        self.validate(config)
        kwargs = {
            "model": config.model,
            "api_key": config.api_key,
        }
        if config.base_url:
            kwargs["base_url"] = config.base_url
        if config.headers:
            kwargs["default_headers"] = config.headers
        if config.temperature is not None:
            kwargs["temperature"] = config.temperature
        return ChatOpenAI(**kwargs)

File: desktop/src/runtime/providers/adapters/openai.ts
import { ChatOpenAI } from "@langchain/openai";
import type { ProviderType } from "@ai-emotion/contracts";
import type { ProviderAdapter, ProviderRuntimeConfig } from "../provider-service";

export class OpenAIAdapter implements ProviderAdapter {
  public readonly providerType: ProviderType = "openai";

  public validate(config: ProviderRuntimeConfig): void {
    if (config.model.trim() === "") {
      throw new Error("openai provider requires model");
    }
    if (!config.api_key || config.api_key.trim() === "") {
      throw new Error("openai provider requires api_key");
    }
  }

  public createModel(config: ProviderRuntimeConfig): ChatOpenAI {
    this.validate(config);

    return new ChatOpenAI({
      model: config.model,
      apiKey: config.api_key,
      configuration: config.base_url ? { baseURL: config.base_url } : undefined,
      defaultHeaders: config.headers ?? undefined,
      temperature: config.temperature ?? undefined
    });
  }
}
